tasks/install.yml

```diff
@@ -1,18 +1,72 @@
 # roles/mariadb/tasks/install.yml
 ---
-
-- name: Add official MariaDB repository
+- name: Add official MariaDB repository Centos/RedHat 7 or Fedora 30
   yum_repository:
     name: MariaDB
     description: Official MariaDB repository
     baseurl: "http://{{ mariadb_mirror }}/{{ mariadb_version }}/{{ ansible_distribution|lower|regex_replace('redhat', 'rhel')|regex_replace('oraclelinux', 'centos') }}{{ ansible_distribution_major_version }}-amd64"
     gpgkey: https://yum.mariadb.org/RPM-GPG-KEY-MariaDB
     gpgcheck: true
   tags: mariadb
+  become: true
+  when:
+    - ansible_os_family == "RedHat"
+    - ansible_facts['distribution_major_version'] == "7" or ansible_facts['distribution_major_version'] == "30"
+
+- name: Check MariaDB Centos/RegHat 8 repo installed
+  stat:
+    path: /etc/yum.repos.d/mariadb.repo
+  register: c8Mariadbpreo
 
-- name: Install packages
+- name: Install Centos/RedHat 8 MariaDB repo
```

Comment:
I now install the .repo file using a template (the `yum_repository:` task causes an issue with EL8. The change you propose, downloading and executing a script, seems more kludgy.